api/server/server.go

```diff
@@ -1123,6 +1126,142 @@ func postContainersCopy(eng *engine.Engine, version version.Version, w http.Resp
 	return nil
 }
 
+// handleArchiveError is a convenience function for the following Copy
+// handlers which makes a JSON error message but leaves writing the
+// header and response to the package's common error handler.
+func handleArchiveError(w http.ResponseWriter, err error) error {
```

Comment:
that is, aside from the `permission denied` change to `httpError()` that I made above.